pkg/client/restclient/request_test.go

```diff
@@ -849,12 +879,20 @@ func TestCheckRetryClosesBody(t *testing.T) {
 			close(ch)
 			return
 		}
-		w.Header().Set("Retry-After", "0")
-		w.WriteHeader(apierrors.StatusTooManyRequests)
+		w.Header().Set("Retry-After", "1")
+		http.Error(w, "Too many requests, please try again later.", apierrors.StatusTooManyRequests)
 	}))
 	defer testServer.Close()
 
+	backoffMgr := &testBackoffManager{t: t, expectedSleeps: make([]time.Duration, 0, 8)}
+	for i := 0; i < 4; i++ {
+		backoffMgr.expectedSleeps = append(backoffMgr.expectedSleeps, time.Duration(0))
+		backoffMgr.expectedSleeps = append(backoffMgr.expectedSleeps, time.Duration(1)*time.Second)
+	}
+	backoffMgr.expectedSleeps = append(backoffMgr.expectedSleeps, time.Duration(0))
```

Comment:
Why not just:

```go
exp := []time.Duration{0, time.Second, 0, time.Second, 0, time.Second, 0, time.Second, 0}
```
?

Also, you call make with a capacity of 8, but append 9 items.